src/main/scala/scalaz/netty/Client.scala

```diff
@@ -36,6 +36,7 @@ import _root_.io.netty.handler.codec._
 
 private[netty] final class Client(limit: Int) {
   // this isn't ugly or anything...
+  @volatile
```

Comment:
Definitely no Guice, but I'd be interested to see a PR to this effect.  I like having `Handler` being scoped, but I'd be willing to sacrifice that to eliminate the var.